src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalHelpers.cs

```diff
@@ -104,6 +105,72 @@ public static string GetRepoPathFromFolder(string sourceFolder, string commit, I
             return null;
         }
 
+        /// <summary>
+        /// Since LibGit2Sharp doesn't support neither sparse checkout not shallow clone
+        /// we implement the flow ourselves.
+        /// </summary>
+        /// <param name="repoUri">The repo to clone Uri</param>
+        /// <param name="branch">The branch to checkout</param>
+        /// <param name="workingDirectory">The working directory</param>
+        /// <param name="logger">The logger</param>
+        /// <param name="remote">The name of the remote</param>
+        /// <param name="user">User name</param>
+        /// <param name="email">User's email</param>
+        /// <param name="pat">User's personal access token</param>
+        /// <param name="repoFolderName">The name of the folder where the repo is located</param>
+        /// <returns>The full path of the cloned repo</returns>
+        public static string SparseAndShallowCheckout(
+            string repoUri,
+            string branch,
+            string workingDirectory,
+            ILogger logger,
+            string remote,
+            string user,
+            string email,
+            string pat,
+            string repoFolderName = "clonedRepo")
+        {
+            Directory.CreateDirectory(workingDirectory);
+
+            ExecuteGitShallowSparseCommand("git", $"init {repoFolderName}", logger, workingDirectory);
+
+            workingDirectory = Path.Combine(workingDirectory, repoFolderName);
+            repoUri = repoUri.Replace("https://", $"https://{user}:{pat}@");
+
+            ExecuteGitShallowSparseCommand("git", $"remote add {remote} {repoUri}", logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("git", "config core.sparsecheckout true", logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("git", "config core.longpaths true", logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("git", $"config user.name {user}", logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("git", $"config user.email {email}", logger, workingDirectory);
+
+            List<string> sparseCommands = new List<string>
+            {
+                "echo eng/ >> .git/info/sparse-checkout",
+                $"echo /{VersionFiles.NugetConfig} >> .git/info/sparse-checkout",
+                $"echo /{VersionFiles.GlobalJson} >> .git/info/sparse-checkout"
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Why create a shell script and do all this work?
I think all of this can be replaced by

```csharp
File.WriteAllLines(Path.Join(workingDirectory, ".git/info/sparse-checkout"), new[]{ "eng/", "/nuget.config", ... })
```

Reply:
So true!